beacon_node/lighthouse_network/src/discovery/mod.rs

```diff
@@ -201,8 +201,24 @@ impl<TSpec: EthSpec> Discovery<TSpec> {
         info!(log, "ENR Initialised"; "enr" => local_enr.to_base64(), "seq" => local_enr.seq(), "id"=> %local_enr.node_id(),
               "ip4" => ?local_enr.ip4(), "udp4"=> ?local_enr.udp4(), "tcp4" => ?local_enr.tcp6()
         );
-
-        let listen_socket = SocketAddr::new(config.listen_address, config.discovery_port);
+        let mut discv5_config = config.discv5_config.clone();
+        let listen_socket = match &config.listen_addresses {
+            crate::config::ListenAddress::V4(v4_addr) => v4_addr.tcp_socket_addr(),
+            crate::config::ListenAddress::V6(v6_addr) => v6_addr.tcp_socket_addr(),
+            crate::config::ListenAddress::DualStack(v4_addr, _v6_addr) => {
+                // TODO: Add two ports to discovery for maximum compatibility.
+                // If that takes too long, change this to use the v6 address with mapped addresses
+                // enabled.
+                v4_addr.tcp_socket_addr()
+            }
+        };
+        if listen_socket.is_ipv6() {
```

Comment:
Isn't creating the `listen_socket` variable unnecessary?
Can't we just do `if config.listen_addresses.v6().is_some() { .. }` ?

Reply:
this is used for starting discv5, however this check was not necessary in the first place